app/assets/javascripts/components/activity/activity_table.cjsx

```diff
@@ -39,9 +43,9 @@ ActivityTable = React.createClass(
 
     drawers = @state.activity.map (revision) ->
       courses = revision.courses.map (course) ->
-        <li><a href="/courses/#{course.slug}">{course.title}</a></li>
+        <li key={Math.random() * course.id}><a href="/courses/#{course.slug}">{course.title}</a></li>
 
-      <tr className='activity-table-drawer'>
+      <tr key={Math.random() * revision.key} className='activity-table-drawer'>
```

Comment:
I agree about the "not right"-ness of it. But what do you mean when you say "But changing the key might be expensive"?